source/common/common/thread.h

```diff
@@ -168,5 +169,19 @@ class AtomicPtr : private AtomicPtrArray<T, 1, alloc_mode> {
   T* get(const MakeObject& make_object) { return BaseClass::get(0, make_object); }
 };
 
+struct MainThread {
+  using MainThreadSingleton = InjectableSingleton<MainThread>;
+  bool inMainThread() const { return main_thread_id_ == std::this_thread::get_id(); }
+  static void init() { MainThreadSingleton::initialize(new MainThread()); }
+  static void clear() {
+    free(MainThreadSingleton::getExisting());
```

Comment:
Post review drive by: I think you have a mismatched new/free here? Should this be delete? cc @chaoqin-li1123

Reply:
Sorry, it is my mistake. i will fix it in a new PR.

Reply:
I guess free doesn't leak memory here because the struct doesn't have a destructor.  In the fixing PR should be changed to new/delete.